platform/broadcom/docker-saiserver-brcm/syncd_init_common.sh

```diff
@@ -0,0 +1,334 @@
+#!/usr/bin/env bash
+
+CMD_SYNCD=/usr/bin/syncd
+
+# dsserve: domain socket server for stdio
+CMD_DSSERVE=/usr/bin/dsserve
+CMD_DSSERVE_ARGS="$CMD_SYNCD --diag"
+
+ENABLE_SAITHRIFT=1
+
+TEMPLATES_DIR=/usr/share/sonic/templates
+PLATFORM_DIR=/usr/share/sonic/platform
+HWSKU_DIR=/usr/share/sonic/hwsku
+
+VARS_FILE=$TEMPLATES_DIR/swss_vars.j2
+
+# Retrieve vars from sonic-cfggen
+SYNCD_VARS=$(sonic-cfggen -d -y /etc/sonic/sonic_version.yml -t $VARS_FILE) || exit 1
+SONIC_ASIC_TYPE=$(echo $SYNCD_VARS | jq -r '.asic_type')
+
+if [ -x $CMD_DSSERVE ]; then
+    CMD=$CMD_DSSERVE
+    CMD_ARGS=$CMD_DSSERVE_ARGS
+else
+    CMD=$CMD_SYNCD
+    CMD_ARGS=
+fi
+
+# Use temporary view between init and apply except when in fast-reboot
+if [[ "$(cat /proc/cmdline)" != *"SONIC_BOOT_TYPE=fast-reboot"* ]]; then
+    CMD_ARGS+=" -u"
+fi
+
+# Use bulk APIs in SAI
+# currently disabled since most vendors don't support that yet
+# CMD_ARGS+=" -l"
+
+# Set synchronous mode if it is enabled in CONFIG_DB
+SYNC_MODE=$(echo $SYNCD_VARS | jq -r '.synchronous_mode')
+if [ "$SYNC_MODE" == "enable" ]; then
+    CMD_ARGS+=" -s"
+fi
+
+case "$(cat /proc/cmdline)" in
+  *SONIC_BOOT_TYPE=fastfast*)
+    if [ -e /var/warmboot/warm-starting ]; then
+        FASTFAST_REBOOT='yes'
+    fi
+    ;;
+  *SONIC_BOOT_TYPE=fast*|*fast-reboot*)
+    # check that the key exists
+    if [[ $(sonic-db-cli STATE_DB GET "FAST_REBOOT|system") == "1" ]]; then
+       FAST_REBOOT='yes'
+    else
+       FAST_REBOOT='no'
+    fi
+    ;;
+  *)
+     FAST_REBOOT='no'
+     FASTFAST_REBOOT='no'
+    ;;
+esac
+
+
+function check_warm_boot()
+{
+    # FIXME: if we want to continue start option approach, then we need to add
+    #        code here to support redis database query.
+    # SYSTEM_WARM_START=`/usr/bin/redis-cli -n 6 hget "WARM_RESTART_ENABLE_TABLE|system" enable`
+    # SERVICE_WARM_START=`/usr/bin/redis-cli -n 6 hget "WARM_RESTART_ENABLE_TABLE|${SERVICE}" enable`
+    # SYSTEM_WARM_START could be empty, always make WARM_BOOT meaningful.
+    # if [[ x"$SYSTEM_WARM_START" == x"true" ]] || [[ x"$SERVICE_WARM_START" == x"true" ]]; then
+    #     WARM_BOOT="true"
+    # else
+        WARM_BOOT="false"
+    # fi
+}
+
+
+function set_start_type()
+{
+    if [ x"$WARM_BOOT" == x"true" ]; then
+        CMD_ARGS+=" -t warm"
+    elif [ x"$FAST_REBOOT" == x"yes" ]; then
+        CMD_ARGS+=" -t fast"
+    elif [ x"$FASTFAST_REBOOT" == x"yes" ]; then
+        CMD_ARGS+=" -t fastfast"
+    fi
+}
+
+config_syncd_cisco_8000()
```

Comment:
Why adding cisco-8000 in platform/broadcom?

Reply:
copied the syncd_init_common.sh from syncd repo.
We will use some other way to copy it from syncd repo

Reply:
ok, copy it automatically.